app_backend/user.py

```diff
@@ -0,0 +1,21 @@
+from fastapi import FastAPI, Response, status
+from dotenv import load_dotenv
+import os
+from database import Database
+
+load_dotenv()
+db = Database(os.getenv("DB_USERNAME"), os.getenv("DB_PASSWORD"))
+db.connect()
+user_db = db.client["user"]
+userinfo_collection = user_db["userlogininfo"]
+resp = Response
+
+class User():
+    def save_user_in_db(loginInfo: dict):
+        for eml in userinfo_collection.find({}, {"_id": 0, "username": 0, "password": 0}):
+            if loginInfo['email'] == eml['email']:
+                resp.status_code = status.HTTP_400_BAD_REQUEST
+                return {"err" : "Email already exists. Use a different email."}
```

Comment:
This `save_user_in_db` function looks like it pulls all users and manually checks all the emails to check for duplicates. Instead, is it possible to query for all entries with that email and then check if the size of our result is greater than 0? If so, that means we found a user with the same email.

This decrease the amount of data communicated by the database and also takes advantage of the quick lookup tables and fancy algorithms MongoDB probably has.

Reply:
Good idea on both the comments you made. There is definetly to do this

Reply:
changed it please check when you can